internal/k8s/reconciler/validator/gateway.go

```diff
@@ -391,7 +391,7 @@ func (g *GatewayValidator) validateTLS(ctx context.Context, state *state.Listene
 		if !errors.As(err, &certificateErr) {
 			return err
 		}
-		state.Status.ResolvedRefs.InvalidCertificateRef = certificateErr
+		state.Status.ResolvedRefs.RefNotPermitted = certificateErr
```

Comment:
```suggestion
		state.Status.ResolvedRefs.InvalidCertificateRef = certificateErr
```
I think reverting this change should fix the expected behavior of the two tests below.